src/transaction/transaction.cpp

```diff
@@ -75,15 +92,30 @@ Transaction::Transaction() :
 void Transaction::sign(Signer const &signer)
 {
     if (m_signature != nullptr)
+    {
         m_signature = nullptr;
+    }
 
-    std::string const txSerialized = Transaction::serialize();
+    std::string const txSerialized = internal::getTxSerializedHashIfRequired(*this);
     std::string const tmpSign = signer.getSignature(txSerialized);
     std::string const signature = util::stringToHex(tmpSign);
 
     m_signature.reset(new std::string(signature));
 }
 
+bool Transaction::verify()
+{
+    if (m_signature == nullptr) throw std::runtime_error(ERROR_MSG_SIGNATURE);
+    if (m_sender == nullptr) throw std::runtime_error(ERROR_MSG_SENDER);
+
+    auto tmp = m_signature;
+    m_signature = nullptr;
+    std::string const txSerialized = internal::getTxSerializedHashIfRequired(*this);
```

Comment:
`getSerialized(asHash: true / false, includeSignature: true / false)` - if this makes sense, perhaps refactor in a future PR?

Reply:
- bool parameter `asHash` wouldn't make sense here, since we already verify inside this method if we should return the hash of the transaction or not, now using function `shouldSignHash`
- added `withSignature` bool parameter here, which made code more clear from the caller side. Good suggestion